src/Features/Core/Portable/AddDebuggerDisplay/AbstractAddDebuggerDisplayCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.AddImports;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.AddDebuggerDisplay
+{
+    internal abstract class AbstractAddDebuggerDisplayCodeRefactoringProvider<TTypeDeclarationSyntax, TMethodDeclarationSyntax> : CodeRefactoringProvider
+        where TTypeDeclarationSyntax : SyntaxNode
+        where TMethodDeclarationSyntax : SyntaxNode
+    {
+        private static readonly SyntaxAnnotation s_trackingAnnotation = new SyntaxAnnotation();
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var type = await context.TryGetRelevantNodeAsync<TTypeDeclarationSyntax>().ConfigureAwait(false);
+
+            if (type is { })
```

Comment:
honestly i'm not a fan.  there are really idiomatic approaches that are clean and simple that i would prefer we stick with.  this is simply `if (type != null)`.   The saving of one char just doesn't feel beneficial to me.  My brain starts having to go "what is this trying to ask?" instead of having it be instantaneously clear.

Reply:
You're right that it's not (yet) as clear. Happy to change.

`== null` and `!= null` are parallel and `is null` and `is { }` are parallels, but they are not the same thing as the first two. Because I've started doing `is null` to be clear that I'm doing ReferenceEquals, `is { }` feels right as a parallel.